File: G_Projects/G_Inventory_Management_System/ProductFactory/ConcreteProducts/ElectronicsProduct.java
package G_Projects.G_Inventory_Management_System.ProductFactory.ConcreteProducts;

import G_Projects.G_Inventory_Management_System.CommonEnums.ProductEnum;
import G_Projects.G_Inventory_Management_System.ProductFactory.Product;

public class ElectronicsProduct extends Product {
    private String brand;
    private int warrantyPeriod; 

    public ElectronicsProduct(String id, String name, double price, int quantity, int threshold) {
        setId(id);
        setName(name);
        setPrice(price);
        setQuantity(quantity);
        setThreshold(threshold);
        setCategory(ProductEnum.ELECTRONICS);
    }

    public String getBrand() {
        return brand;
    }
    public void setBrand(String brand) {
        this.brand = brand;
    }

    public int getWarrantyPeriod() {
        return warrantyPeriod;
    }
    public void setWarrantyPeriod(int warrantyPeriod) {
        this.warrantyPeriod = warrantyPeriod;
    }
}

File: Python/A_Project/D_Inventory_Management_System/ProductFactory/ConcreteProducts/electronics_product.py
from CommonEnums.product_enum import ProductEnum
from ProductFactory.product import Product


class ElectronicsProduct(Product):
    def __init__(self, product_id, name, price, quantity, threshold):
        super().__init__()
        self._brand = None
        self._warranty_period = 0
        self.set_id(product_id)
        self.set_name(name)
        self.set_price(price)
        self.set_quantity(quantity)
        self.set_threshold(threshold)
        self.set_category(ProductEnum.ELECTRONICS)

    def get_brand(self):
        return self._brand

    def set_brand(self, brand: str):
        self._brand = brand

    def get_warranty_period(self):
        return self._warranty_period

    def set_warranty_period(self, warranty_period: int):
        self._warranty_period = warranty_period
